jenkins/ci.suse.de/openstack-submit-project.yaml

```diff
@@ -25,90 +25,83 @@
+                    exit 0
+              ;;
           esac
 
+          BS=IBS
+          OSC="osc -A https://api.suse.de"
 
           COS=$project
           ECOS=""
           COSS=$COS:$subproject
-          export BS_CHECKOUT=/home/jenkins/${BS}_CHECKOUT/$COSS
+          BS_CHECKOUT=/home/jenkins/${BS}_CHECKOUT/$COSS
           submitcmd="$OSC copypac -K -e"
           OSCBUILDDIST=SLE_12_SP1
 
           case $project in
-          Devel:Cloud:6)
-              ECOS=systemsmanagement:crowbar:3.0
-          ;;
-          Devel:Cloud:6:SAP)
-              ECOS=systemsmanagement:crowbar:3.1
-          ;;
-          Devel:Cloud:7)
-              OSCBUILDDIST=SLE_12_SP2
-              ECOS=systemsmanagement:crowbar:4.0
-          ;;
-          Devel:Cloud:8)
-              OSCBUILDDIST=SLE_12_SP3
-              ECOS=systemsmanagement:crowbar:5.0
-          ;;
-          *) echo "Please add support for the project: $project"
-              exit 1
+              Devel:Cloud:6)
+                  ECOS=systemsmanagement:crowbar:3.0
+              ;;
+              Devel:Cloud:6:SAP)
+                  ECOS=systemsmanagement:crowbar:3.1
+              ;;
+              Devel:Cloud:7)
+                  OSCBUILDDIST=SLE_12_SP2
+                  ECOS=systemsmanagement:crowbar:4.0
+              ;;
+              Devel:Cloud:8)
+                  OSCBUILDDIST=SLE_12_SP3
+                  ECOS=systemsmanagement:crowbar:5.0
+              ;;
+              *) echo "Please add support for the project: $project"
+                  exit 1
           esac
 
+          pkg_filter=`echo ${packagefilter}`
+          pkg_filter="${pkg_filter// /|}"
+          for component in `$OSC ls $COSS 2>/dev/null | grep -E "^(${pkg_filter})" | grep -v "^_product."` ; do
+              cd $BS_CHECKOUT
+              if [[ ! -d $component ]] ; then
+                  $OSC co -o $component $COSS $component
+              fi
+              cd $component
 
-          for COMPONENT in `$OSC ls $COSS 2>/dev/null | grep -E "^(${packagefilter// /|})" | grep -v "^_product."` ; do
-          cd $BS_CHECKOUT || exit 1
-          if [ ! -d $COMPONENT ] ; then
-              $OSC co -o $COMPONENT $COSS $COMPONENT
-          fi
-          cd $COMPONENT || exit 1
+              if ! grep -q "<linkinfo" .osc/_files ; then
+                  echo "Error: package $component in $subproject (staging) is not a link to non-staging."
+                  exit 2
+              fi
 
-          grep -q "<linkinfo" .osc/_files || exit 2
+              # avoid copying untested versions
+              lastchange=$($OSC api "source/$COSS/$component/_history" | perl -ne 'if(m/<time>(\d+)</) {print $1; exit 0}')
+              if [[ $lastchange -gt $starttime ]] ; then
+                  echo "Skipping $component because due to a change after the test finished. test time: $starttime :: last change: $lastchange"
```

Comment:
`because due to` doesnt sound like correct English.
and $starttime is meant to be the time the test started, not `finished` because if we start a test, then something changes and then we finish, we must not submit it. (there will be another testrun later)

Reply:
will fix the wording,
and the starttime is the time that previous job finished - the submit job is following the actual "test job"